app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -46,11 +81,36 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
             return value;
         }
 
-        function removeFilter(filterKey, value, $event) {
+        function removeFilter(filterKey, value, savedSearch, $event) {
             $event.preventDefault();
             $event.stopPropagation();
-            PostFilters.clearFilter(filterKey, value);
+            if (savedSearch) {
+                savedSearch.filter = PostFilters.clearFilterFromArray(filterKey, value, savedSearch.filter);
+                PostFilters.clearFilter(filterKey, value);
+                PostFilters.setFilter('saved_search', savedSearch);
+                $scope.savedSearch = savedSearch;
+            } else {
+                PostFilters.clearFilter(filterKey, value);
+            }
         }
+        function removeSavedSearch(savedSearch, $event) {
+            $event.preventDefault();
```

Comment:
this whole function is a bit silly . I'll look into simplifying this removeFilter/removeSavedSearch feature if possible